### PR TITLE
BoxPokemon struct cleanup

### PR DESCRIPTION
**Pull request by AsparagusEduardo**
Requires #122 to be merged first.

- Renamed `BOXMON` to `BoxPokemon`.
- Standarized pointer star positions for `BoxPokemon` pointers.
- Standarized most variable names using `BoxPokemon`.

AsparagusEduardo#6051

### CODE REVIEW

**Comment by AsparagusEduardo** [comment]
Ready for review

File: include/pokemon_types_def.h [hunk 1]
```diff
@@ -116,15 +116,15 @@ typedef union {
     PokemonDataBlockD blockD;
 } PokemonDataBlock;
 
-typedef struct BoxPokemon {
+typedef struct {
```
Comment (red031000) [inline include/pokemon_types_def.h:119]: should have BoxPokemon here after the typedef struct

**Review by red031000 — CHANGES_REQUESTED** [review]
only one issue

**Comment by AsparagusEduardo** [comment]
Done

**Review by red031000 — APPROVED** [review]

**Pull request merged**